src/modules/user/services/find-all-users.service.ts

```diff
@@ -1,17 +1,16 @@
 import { Injectable } from '@nestjs/common';
+import { UserEntity } from 'src/database/entities/users.entity';
+import { PageDto, PageOptionsDto } from 'src/shared/pagination';
+
 import { UserRepository } from '../repository/user.repository';
 
 @Injectable()
 export class FindAllUsersService {
   constructor(private userRepository: UserRepository) {}
 
-  async execute() {
-    const users = await this.userRepository.getAllUsers();
-
-    if (users.length <= 0) {
-      return { message: 'Users is empty' };
-    }
+  async execute(pageOptionsDto: PageOptionsDto): Promise<PageDto<UserEntity>> {
```

Comment:
o que acha de retornar a promise nesse caso?
a menos que for fazer alguma validação no futuro